File: client/src/types/schedule/schedule.ts
/**
 * The weekdays a user can mark as their default in-office days during signup.
 *
 * @TODO: Persist the selected default schedule to the API once an endpoint is
 * available.
 */
export interface Weekday {
  /** Stable identifier sent to the API when saving the user's schedule. */
  id: string;
  /** Abbreviated label shown on the card (e.g. "Mon"). */
  label: string;
}

export const WEEKDAYS: Weekday[] = [
  { id: "monday", label: "Mon" },
  { id: "tuesday", label: "Tue" },
  { id: "wednesday", label: "Wed" },
  { id: "thursday", label: "Thu" },
  { id: "friday", label: "Fri" },
];

File: api/app/models/schedule.rb
class Schedule < ApplicationRecord
  DAYS = %w[
    monday
    tuesday
    wednesday
    thursday
    friday
    saturday
    sunday
  ]

  belongs_to :user

  validates :is_default,
    inclusion: { in: [ true, false ] }

  DAYS.each do |day|
    validates day,
      inclusion: { in: [ true, false ] }
  end

  scope :default, -> { where(is_default: true) }
end
